api/provider/aws/task_list.go

```diff
@@ -0,0 +1,151 @@
+package aws
+
+import (
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/quintilesims/layer0/common/config"
+	"github.com/quintilesims/layer0/common/errors"
+	"github.com/quintilesims/layer0/common/models"
+)
+
+const MAX_TASK_IDS = 100
+
+func (t *TaskProvider) List() ([]models.TaskSummary, error) {
+	clusterNames, err := t.listClusterNames()
+	if err != nil {
+		return nil, err
+	}
+
+	taskCopies := map[string][]*ecs.Task{}
+	for _, clusterName := range clusterNames {
+		environmentID := delLayer0Prefix(t.Config.Instance(), clusterName)
```

Comment:
I don't think this is correct. You should use the fully-qualified environment id in aws calls.

Reply:
This and the previous comment came after I started to consider just returning the cluster ARNs from AWS rather than getting their names. Do you have a preference here whether I should use the prefix-included cluster name or the ARN?

Reply:
I'd recommend moving the [existing function](https://github.com/quintilesims/layer0/blob/api-refactor/api/provider/aws/environment_list.go#L34) into `common.go` and just using that